streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedStream.java

```diff
@@ -381,7 +381,8 @@
      * {@link StreamsConfig#COMMIT_INTERVAL_MS_CONFIG commit intervall}.
      *
      * <p>
-     * For failure and recovery the store will be backed by an internal changelog topic that will be created in Kafka.
+     * For failure and recovery the store (which always will be of type {@link TimestampedKeyValueStore}) will be backed by
+     * an internal changelog topic that will be created in Kafka.
```

Comment:
Should apply the same improvement to `reduce()` and `count()` overloads? Also for `CogroupedKStream#aggregate()`?

What about `TimeWindowedKStream` and `TimeWindowedCogroupedKStream` ?

Also `StreamsBuilder#table()` (and `#globalTable()`) might need an update?

Reply:
Yes, this should be probably done. But since I don't know how they work and what to write there, I'd prefer to do it in another PR (or someone else should do it).

Reply:
`reduce()` and `count()` are similar to `aggregate()`: reduce only does not allow you to change the value type (ie, output type == input type) and count, well implements an aggregate() that counts :)

And they all work the same for `KGroupedStream`, `CogroupedKStream`, `TimeWindowedKStream` and `TimeWindowedCoGroupedKStream`.

`table()` and `globalTable()` just read a topic and upsert the data into a state store / table.

So it would be great to do them all in one PR?

Reply:
I'd prefer to commit this chages and create an issue referring to this PR. What do you think?

Reply:
We can still do multiple PRs -- feel free to open them in parallel reusing the existing JIRA ticket.

Will hold off to merge this because if we can detect other things we need to change in the other PRs, we can keep the applied changes in-sync more easily.